README.md

```diff
@@ -1,218 +1,207 @@
+### Connecting
 
-If you want to send and receive messages to the embedded element you need to call `connect()` which returns a Promise that resolves to the communication interface of the given element:
+Once the client is built, call connect to create the IFRAME. The connect process creates the `src` attribute used for the actual IFRAME. How the `src` value is generated is based upon how the embed SDK is initialized.
+
+1. Signed - the endpoint specified by the 2nd argument of the `init` call is called. The endpoint is expected to return a signed embed login URL.
+2. Cookieless - the endpoint or the function specified by the 2nd argument of the `initCookieless` call is called. The endpoint or function is expected to return cookieless tokens, in particular the authentication and navigation tokens. The tokens are appended to the embed login URL.
+3. Private - the embed connection is private if the 2nd argument of the `init` call is not provided. In this case the URL is derived from the builder and decorated with the parameters required for Looker embed. For private embed, it is expected that the user is already logged into Looker OR that embedding URL include the `allow_login_screen=true` parameter.
+
+`connect` returns a Promise that resolves to the connection interface for the embedded IFRAME.
 
 ```javascript
   .connect()
-  .then(setupDashboard)
+  .then((connection) => {
+    // Save the connection
+  })
   .catch(console.error)
 ```
 
-## Building URLs for the SDK
-
-This section does not apply to cookieless embed as building URLs in this manner is not supported. See the [cookieless embed](#cookieless) section for details.
+### Interacting
 
-The main documentation for Looker SSO embed URLs is [here](https://docs.looker.com/r/sdk/sso-embed). The only difference when creating URLs for the SDK is that you will need to add an `sdk=2` parameter to the Embed URL alongside other parameters like filters and the `embed_domain` parameter. This parameter allows Looker to identify that the SDK is present and can take advantage of additional features provided by the SDK.
+Embed SDK 2.0.0 returns a unified connection that supports interacting with all Looker content types. The Embedding application can determine what kind of content is currently being displayed and interact accordingly.
 
-```html
-/embed/looks/4?embed_domain=https://mywebsite.com =>
-/embed/looks/4?embed_domain=https://mywebsite.com&sdk=2
+```javascript
+if (connection.getPageType() === 'dashboards') {
+  connection.asDashboardConnection().run()
+} else (connection.getPageType() === 'looks') {
+  connection.asLookConnection().run()
+} else (connection.getPageType() === 'explore') {
+  connection.asExploreConnection().run()
```

Comment:
do you want to add reports and visualizations to the snippet for completeness?

Reply:
Reports do not support the run option as yet